File: Camera/app/app/src/main/java/com/bi2qfa/sonyconnect/Rand.java
package com.bi2qfa.sonyconnect;

import java.security.SecureRandom;










public final class Rand {

    private static final SecureRandom RNG = new SecureRandom();

    private Rand() {
    }

    
    public static byte[] bytes(int n) {
        byte[] out = new byte[n];
        RNG.nextBytes(out);
        return out;
    }
}

File: Phone/app/src/main/java/com/bi2qfa/sonyconnect/crypto/Rand.kt
package com.bi2qfa.sonyconnect.crypto

import java.security.SecureRandom

/**
 * 唯一的随机源：本机 8 字节设备码。
 *
 * <p>原来它是 `KeyDerivation.random` —— 整套加解密（AES-CBC + HMAC、会话密钥
 * 派生、PBKDF2、双向认证）已经整体移除。随机源本身与加解密无关：设备码必须是
 * **不可预测**的随机值，所以单独留这么一个小对象。相机端有对称的 `Rand.java`。
 */
object Rand {

    private val RNG = SecureRandom()

    fun bytes(n: Int): ByteArray {
        val out = ByteArray(n)
        RNG.nextBytes(out)
        return out
    }
}
